app/components/join-steps/step-three.js

```diff
@@ -26,12 +27,16 @@ export default class StepThreeComponent extends Component {
   }
 
   @action inputHandler(e) {
+    console.log('I am ruuning');
     this.setIsPreValid(false);
-    this.data = { ...this.data, [e.target.name]: e.target.value };
-    localStorage.setItem('stepThreeData', JSON.stringify(this.data));
-    const validated =
-      validator(this.data.whyRds, 100) && validator(this.data.foundFrom, 1);
-    this.setIsValid(validated);
-    localStorage.setItem('isValid', validated);
+    const setValToLocalStorage = () => {
+      this.data = { ...this.data, [e.target.name]: e.target.value };
+      localStorage.setItem('stepThreeData', JSON.stringify(this.data));
+      const validated =
+        validator(this.data.whyRds, 100) && validator(this.data.foundFrom, 1);
+      this.setIsValid(validated);
+      localStorage.setItem('isValid', validated);
+    };
+    debounce(this.data, setValToLocalStorage, 1000);
```

Comment:
We can maybe use a `constant` for the debounce timer

Reply:
added constant✅